### PR TITLE
Add project around file parsing

### PR DESCRIPTION
**Pull request by illicitonion**
Fixes https://github.com/CodeYourFuture/immersive-go-course/issues/82

Sample implementation in https://github.com/CodeYourFuture/immersive-go-course/tree/impl/file-parsing (https://github.com/CodeYourFuture/immersive-go-course/pull/115)

### CODE REVIEW

File: projects/file-parsing/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+json.txt contains an array of JSON objects, each with a "name" and "high_score" key.
+
+### Repeated JSON
+
+repeated-json.txt contains lines of data stored in JSON format. Each line contains exactly one record, stored as an object. Lines starting with # are comments and should be ignored.
+
+### CSV
+
+data.csv is a standard CSV file. The format is well-documented online, and there are many libraries which support parsing it.
```
Comment (sre-is-laura) [inline projects/file-parsing/README.md:34]: IMO worth spelling out Comma Separated Value on first use

File: projects/file-parsing/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+<!--forhugo
++++
+title="File Parsing"
++++
+forhugo-->
+
+In this project we'll practice parsing data from files in different formats. Often times it's convenient to write a small program to understand or process some data, and being comfortable quickly doing so can be very helpful.
```
Comment (sre-is-laura) [inline projects/file-parsing/README.md:7]: Additionally, larger projects often require reading configuration or data files.

Comment (sre-is-laura) [reply projects/file-parsing/README.md:7]: (suggested addition)

File: projects/file-parsing/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+repeated-json.txt contains lines of data stored in JSON format. Each line contains exactly one record, stored as an object. Lines starting with # are comments and should be ignored.
+
+### CSV
+
+data.csv is a standard CSV file. The format is well-documented online, and there are many libraries which support parsing it.
+
+### Custom Binary
+
+There are two files in a custom binary serialisation format. The format is as follows:
+* First two bytes of the file indicate endianness of numbers. If the bytes are FE FF, numbers in the file are stored in big endian byte order. If the bytes are FF FE, numbers in the file are stored in little endian byte order.
```
Comment (sre-is-laura) [inline projects/file-parsing/README.md:39]: you may need to explain the endianness concept here

File: projects/file-parsing/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+* Each record contains exactly four bytes representing the score as a signed 32-bit integer, in the above described endian format, then the name of the player stored in UTF-8 which may not contain a null character, followed by a null terminating character.
```
Comment (sre-is-laura) [inline projects/file-parsing/README.md:40]: you may also need to explain UTF 8 and terminating characters here. i don't think we can assume people have had the experience of wrangling C style strings

File: projects/file-parsing/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
++++
+forhugo-->
+
+In this project we'll practice parsing data from files in different formats. Often times it's convenient to write a small program to understand or process some data, and being comfortable quickly doing so can be very helpful.
+
+Timebox: 2 days
+
+## Objectives:
+
+- Write code to parse data in non-standard formats.
```
Comment (sre-is-laura) [inline projects/file-parsing/README.md:13]: should we specify somewhere that we expect Go implementations?
i wonder as well: should we ask them to practice their shell by doing the same tasks using command line scripting utilities (sed/awk/jq and so on). can be a really useful skill for troubleshooting. at least for the jsons and csvs, the binary formats might be a bit much

Comment (sre-is-laura) [reply projects/file-parsing/README.md:13]: also maybe explicitly suggest writing unit tests, although TBF i don't think we did for the rest of the projects

File: projects/file-parsing/README.md (hunk 4 above)
Comment (illicitonion) [reply projects/file-parsing/README.md:53]: Good call - added some definitions, and then a section to consider trade-offs of different encodings of variable length data (see also https://github.com/CodeYourFuture/immersive-go-course/issues/112)

**Review by sre-is-laura — APPROVED** [review]

**Pull request merged**